bleak/backends/winrt/client.py

```diff
@@ -0,0 +1,824 @@
+                        CONTROLLER_ERROR_CODES.get(
+                            write_result.protocol_error, "Unknown"
+                        ),
+                    )
+                )
+            else:
+                raise BleakError(
+                    "Could not write value {0} to descriptor {1}: {2}".format(
+                        data,
+                        descriptor.uuid,
+                        _communication_statues.get(write_result.status, ""),
+                    )
+                )
+
+    async def start_notify(
+        self,
+        char_specifier: Union[BleakGATTCharacteristic, int, str, uuid.UUID],
+        callback: Callable[[int, bytearray], None],
+        **kwargs
+    ) -> None:
+        """Activate notifications/indications on a characteristic.
+
+        Callbacks must accept two inputs. The first will be a uuid string
+        object and the second will be a bytearray.
+
+        .. code-block:: python
+
+            def callback(sender, data):
+                print(f"{sender}: {data}")
+            client.start_notify(char_uuid, callback)
+
+        Args:
+            char_specifier (BleakGATTCharacteristic, int, str or UUID): The characteristic to activate
+                notifications/indications on a characteristic, specified by either integer handle,
+                UUID or directly by the BleakGATTCharacteristic object representing it.
+            callback (function): The function to be called on notification.
+
+        """
+        if inspect.iscoroutinefunction(callback):
+
+            def bleak_callback(s, d):
+                asyncio.ensure_future(callback(s, d))
+
+        else:
+            bleak_callback = callback
+
+        if not isinstance(char_specifier, BleakGATTCharacteristic):
+            characteristic = self.services.get_characteristic(char_specifier)
+        else:
+            characteristic = char_specifier
+        if not characteristic:
+            raise BleakError("Characteristic {0} not found!".format(char_specifier))
+
+        if self._notification_callbacks.get(characteristic.handle):
+            await self.stop_notify(characteristic)
+
+        characteristic_obj = characteristic.obj
+        if (
+            characteristic_obj.characteristic_properties
+            & GattCharacteristicProperties.INDICATE
+        ):
+            cccd = GattClientCharacteristicConfigurationDescriptorValue.INDICATE
+        elif (
+            characteristic_obj.characteristic_properties
+            & GattCharacteristicProperties.NOTIFY
+        ):
+            cccd = GattClientCharacteristicConfigurationDescriptorValue.NOTIFY
+        else:
+            cccd = GattClientCharacteristicConfigurationDescriptorValue.NONE
+
+        fcn = _notification_wrapper(bleak_callback, asyncio.get_event_loop())
+        fcn_token = characteristic_obj.add_value_changed(fcn)
+        self._notification_callbacks[characteristic.handle] = fcn, fcn_token
```

Comment:
Do we actually need to include `fcn` here? I don't see where it is used later anywhere.

Also `event_handler_token` might be a better name than `fcn_token` to indicate that this is a .NET event handler we are talking about.

Reply:
No, it is not needed. I had it there so I could call `del` on it previously. This to try to remedy all possible memory issues...
Will try to remove this.